### PR TITLE
Integrate provider asset_urls endpoint

### LINKED ISSUE #607: Integrate new exposed asset URL(s) (V3) from Provider
Linked Provider Issue: https://github.com/oceanprotocol/provider/issues/281

### PR DESCRIPTION
**Pull request by maldo-ctrl**
Signed-off-by: Pablo Maldonado <pablo@oceanprotocol.com>

Fixes #607  .

Changes proposed in this PR:

- Integrate asset_url endpoint in DataServiceProvider

### CODE REVIEW

**Review by calina-c — COMMENTED** [review]
Asked a small question. LGTM otherwise.

File: ocean_lib/data_provider/data_service_provider.py [hunk 1]
```diff
@@ -812,6 +818,35 @@ def check_asset_file_info(did: str, provider_uri: str) -> bool:
 
         return False
 
+    @staticmethod
+    @enforce_types
+    def get_asset_urls(
+        did: str, service_id: str, provider_uri: str, wallet: Wallet
+    ) -> None:
+        if not did:
+            return False
```
Comment (DMats) [inline ocean_lib/data_provider/data_service_provider.py:827]: This error returns `False`. but the error on line 846 returns `None`.  Please pick one error convention.

File: ocean_lib/data_provider/data_service_provider.py [hunk 2]
```diff
@@ -812,6 +818,35 @@ def check_asset_file_info(did: str, provider_uri: str) -> bool:
 
         return False
 
+    @staticmethod
+    @enforce_types
+    def get_asset_urls(
+        did: str, service_id: str, provider_uri: str, wallet: Wallet
+    ) -> None:
```
Comment (DMats) [inline ocean_lib/data_provider/data_service_provider.py:825]: Once you've selected an error convention, update this return type hint.

```
Optional[dict]
```
or
```
Union[bool, dict]
```

**Review by DMats — CHANGES_REQUESTED** [review]

**Review by calina-c — APPROVED** [review]
Good work! Did you merge v4main? Let me know if I can merge :)

**Pull request merged**